### PR TITLE
Log config on lookoutd serve startup

### PR DESCRIPTION
**Pull request by carlosms**
Part of #181.

DB config left intentionally out, as it may contain the password.
I'm initializing the structs this way to avoid changing a pointer reference accidentally, and to avoid leaking by mistake any future fields that may contain credentials.

The output will be similar to this:
```
[2018-09-04T14:57:52.179461203+01:00]  INFO starting lookoutd conf=main.Config{
  Config: server.Config{
    Analyzers: []lookout.AnalyzerConfig{
      lookout.AnalyzerConfig{
        Name: "Dummy",
        Addr: "ipv4://localhost:10302",
        Disabled: false,
        Feedback: "https://github.com/carlosms-bot/analyzer-comments/issues/new?template=feedback.md",
        Settings: map[string]interface {}{
        },
      },
    },
  },
  Providers: struct { Github github.ProviderConfig }{
    Github: github.ProviderConfig{
      CommentFooter: "_If you have feedback about this comment please, [tell us](%s)._",
    },
  },
  Repositories: []main.RepoConfig{
    main.RepoConfig{
      URL: "https://github.com/carlosms/lookout-test",
      Client: github.ClientConfig{
        User: "",
        Token: "****",
        MinInterval: "",
      },
    },
  },
} options=main.ServeCommand{
  CommonOptions: cli.CommonOptions{
    LogOptions: cli.LogOptions{
      LogLevel: "DEBUG",
      LogFormat: "",
      LogFields: "",
      LogForceFormat: false,
      Verbose: false,
    },
    GrpcMaxMsgSize: 100,
  },
  DBOptions: cli.DBOptions{
    DB: "",
  },
  ConfigFile: "config.yml",
  GithubUser: "",
  GithubToken: "****",
  DataServer: "ipv4://localhost:10301",
  Bblfshd: "ipv4://localhost:9432",
  DryRun: true,
  Library: "/tmp/lookout",
  Provider: "github",
  Positional: struct { Repository string "positional-arg-name:\"repository\"" }{
    Repository: "https://github.com/carlosms/lookout-test",
  },
} source=lookoutd/serve.go:175

```

### CODE REVIEW

**Comment by carlosms** [comment]
Lookout failed with this error. It looks to me like it's because of #199.
 
```json
{
  "app":"lookoutd",
  "error":"posting analysis failed: github api error: POST https://api.github.com/repos/src-d/lookout/pulls/220/reviews: 422 Validation Failed [{Resource: Field: Code: Message:}]",
  "event-id":"fe231036afe456a7be94d814a1ccd03325d0cabc",
  "event-type":2,
  "level":"error",
  "msg":"event processing failed",
  "source":"server/server.go:101",
  "time":"2018-09-04T14:05:11.99550047Z"
}
```

**Comment by bzz** [comment]
SGTM, but would not these logs contain a plenty of newlines?

If we are to us  `master` of litter instead, we should be able to use `litter.Config.Compact = true` to make it line-oriented again.

File: cmd/lookoutd/serve.go [hunk 1]
```diff
@@ -140,6 +143,39 @@ func (c *ServeCommand) Execute(args []string) error {
 	return server.NewServer(watcher, poster, dataHandler.FileGetter, analyzers, eventOp, commentsOp).Run(ctx)
 }
 
+func (c *ServeCommand) logConfig(conf Config) {
+	cCp := ServeCommand{
+		ConfigFile:    c.ConfigFile,
```
Comment (smacker) [inline cmd/lookoutd/serve.go]: I'm pretty sure sooner or later somebody (read: me) will forget to update the struct here when changing ServeCommand.

How about just to deep-copy it and rewrite necessary fields?
You can use this lib for it for example: https://github.com/jinzhu/copier

Comment (carlosms) [reply cmd/lookoutd/serve.go]: I think we have to make a compromise here. We either risk forgetting to add new fields, or we risk forgetting to remove new fields with sensitive info.

I thought I'd go with the safest way. But I could change it for the deep copy if you guys think that would be better. We probably won't be adding many fields with credentials or secrets.

Comment (smacker) [reply cmd/lookoutd/serve.go]: I can see only 2 such new field: private key and queue connection string. But we will definetely have more configuration options. The thing is: it's easy to spot sensetive fields in logs and fix it. Not that easy to spot missing fields.

**Comment by carlosms** [comment]
It's true we output a lot of lines you may not be always interested on, but on the other hand that makes it readable. I personally prefer that verbosity to this compact output:

```
[2018-09-05T13:20:48.486148115+01:00]  INFO starting lookoutd conf=main.Config{C
onfig:server.Config{Analyzers:[]lookout.AnalyzerConfig{lookout.AnalyzerConfig{Na
me:"Dummy",Addr:"ipv4://localhost:10302",Disabled:false,Feedback:"https://github
.com/carlosms-bot/analyzer-comments/issues/new?template=feedback.md",Settings:ma
p[string]interface{}{}}}},Providers:struct{Github github.ProviderConfig}{Github:
github.ProviderConfig{CommentFooter:"_If you have feedback about this comment pl
ease, [tell us](%s)._"}},Repositories:[]main.RepoConfig{main.RepoConfig{URL:"htt
ps://github.com/carlosms/lookout-test",Client:github.ClientConfig{User:"",Token:
"****",MinInterval:""}}}} options=main.ServeCommand{CommonOptions:cli.CommonOpti
ons{LogOptions:cli.LogOptions{LogLevel:"info",LogFormat:"",LogFields:"",LogForce
Format:false,Verbose:false},GrpcMaxMsgSize:100},DBOptions:cli.DBOptions{DB:""},C
onfigFile:"config.yml",GithubUser:"",GithubToken:"****",DataServer:"ipv4://local
host:10301",Bblfshd:"ipv4://localhost:9432",DryRun:true,Library:"/tmp/lookout",P
rovider:"github",Positional:struct{Repository string "positional-arg-name:\"repo
sitory\""}{Repository:"https://github.com/carlosms/lookout-test"},analyzers:map[
string]pb.AnalyzerClient{},pool:nil}
```

But I could also go with it if you feel strongly about it.

**Comment by bzz** [comment]
No strong preference. Do you think multiline output will not confuse the log collection then? That's the only thing I had in mind

**Comment by carlosms** [comment]
With the log format set to json, as we do for staging, it will just be a string with a bunch of "\n".
But just in case we end up with a system that needs each log in one line, I changed it in dc61fff.

File: cmd/lookoutd/serve.go (hunk 1 above)
Comment (carlosms) [reply cmd/lookoutd/serve.go]: That's the problem, sensitive fields should not be easy to spot in the logs :D
Anyway, I made the change in cbe937d, let's keep an eye on new fields to make sure we don't leak anything.

Comment (smacker) [reply cmd/lookoutd/serve.go]: haha. I mean it's easy to spot during development. You would see it when you run the server locally and kill it.

**Comment by carlosms** [comment]
rebased, squashed, and merged as soon as CI is green

**Pull request merged**